Lib/test/test_mailbox.py

```diff
@@ -532,6 +532,11 @@ def _test_flush_or_close(self, method, should_call_close):
             self.assertIn(self._box.get_string(key), contents)
         oldbox.close()
 
+    def test_use_context_manager(self):
+        # Mailboxes are usable as a context manager
+        with self._factory(self._path) as box:
+            self.assertIsInstance(box, self._box.__class__)
+
```

Comment:
Could you add a test for the lock?

Reply:
@matrixise I'm unsure how to do it right. For example:
- `Maildir` doesn't use the lock. So the test is not useful there.
- `Mailbox` uses it and store the state in an internal boolean variable (`self._lock`). So the test would check the implementation.

It would be nice to have a `locked` attribute for `Mailbox` but it will not work for each case and is probably too invasive for this patch.

Reply:
good question, I don't remember your code, but I am going to assign myself your PR.

Reply:
Ok, thank you!